src/quacc/calculators/orca/orca.py

```diff
@@ -0,0 +1,193 @@
+    """
+
+    def __init__(self, **kwargs):
+        self.kwargs = kwargs
+        profile = kwargs.get("profile", None)
+        directory = kwargs.get("directory", ".")
+        parameters = kwargs.copy()
+        parameters.pop("profile", None)
+        parameters.pop("directory", None)
+
+        def write_input(inner_self, directory, atoms, parameters, properties):
+            parameters = dict(parameters)["parameters"]
+
+            kw = dict(
+                charge=0,
+                mult=1,
+                orcasimpleinput="B3LYP def2-TZVP",
+                orcablocks="%pal nprocs 1 end",
+            )
+            kw.update(parameters)
+
+            write_orca(directory / inner_self.input_file, atoms, kw)
+
+        OrcaTemplate.write_input = write_input
+
+        super().__init__(
+            template=OrcaTemplate(),
+            profile=profile,
+            directory=directory,
+            parameters=parameters,
+        )
+
+
+@writer
+def write_orca(fd, atoms, params):
+    """
+    Function to write out the ORCA input file. This is a modified version of the
+    write_input function in the ASE ORCA calculator. It allows for the use of
+    pseudo-hydrogens to represent dangling bonds.
+
+    Parameters
+    ----------
+    fd : file descriptor
+        File descriptor to write to
+    atoms : Atoms
+        ASE Atoms object
+    params : dict
+        Dictionary of parameters to write out for ORCA. This can be charge, mult, orcasimpleinput, orcablocks.
+    """
+
+    # Write out the orca simple input and the orca blocks
+    fd.write("! %s \n" % params["orcasimpleinput"])
+    fd.write("%s \n" % params["orcablocks"])
+
+    # If the atoms object has the arrays "oxi_states" and "atom_type" then we
+    # need to add the point charges with the corresponding oxi_state value as
+    # its charge and H> to create an embedding potential (with ghost basis)
+    if "oxi_states" in atoms.arrays and "atom_type" in atoms.arrays:
+        fd.write("%coords\n")
+        fd.write("CTyp xyz\n")
+        fd.write("Units angs\n")
+        fd.write("coords\n")
+        # Sum up cluster charge from the charges on the dangling bonds
+        cluster_charge = 0
+        oxi_states = atoms.get_array("oxi_states")
+        atom_type = atoms.get_array("atom_type")
+        for index, atom in enumerate(atoms):
+            # If atom_type is 'pH', then it is a dangling bond
+            if atom_type[index] == "pH":  # 71 is ascii G (Ghost)
+                cluster_charge += oxi_states[index]
+                symbol = atom.symbol + ">  "
+                fd.write(
+                    symbol
+                    + "  "
+                    + str(oxi_states[index])
+                    + "  "
+                    + str(atom.position[0])
+                    + "  "
+                    + str(atom.position[1])
+                    + "  "
+                    + str(atom.position[2])
+                    + "\n"
```

Comment:
Use f-strings instead of + signs. f"{oxi_states[index}   {atom.position[0]}   blah blah"